service/storage_service.cc

```diff
@@ -5278,7 +5278,7 @@ future<> storage_service::load_tablet_metadata(const locator::tablet_metadata_ch
     }, acquire_merge_lock::no);
 }
 
-future<> storage_service::process_tablet_split_candidate(table_id table) {
+future<> storage_service::process_tablet_split_candidate(table_id table) noexcept {
```

Comment:
find_column_family() below can throw, and I'm sure split_all_storage_groups() can too. Slapping noexcept on a function isn't enough.

Reply:
I suggest to keep it possibly throwing and follow  up with an exception safety fix (which won't be simple).

Reply:
We handle exceptions and don't propagate it, in process_tablet_split_candidate(). All the calls outside the try catch block, are already noexcept. the reason I added noexcept is to document the behavior, as it's important that the split monitor keeps running despite any exception.

Reply:
Ah, ok. Only the definition of the lambda calling find_column_family is outside the try block, not its execution.